src/app/InteractionModelEngine.cpp

```diff
@@ -93,10 +93,10 @@ bool MayHaveAccessibleEventPathForEndpoint(DataModel::Provider * aProvider, Endp
                                                                aSubjectDescriptor);
     }
 
-    for (auto & cluster : aProvider->ServerClustersIgnoreError(aEventPath.mEndpointId))
+    for (auto & cluster : aProvider->ServerClustersIgnoreError(aEndpoint))
     {
-        if (MayHaveAccessibleEventPathForEndpointAndCluster(ConcreteClusterPath(aEventPath.mEndpointId, cluster.clusterId),
-                                                            aEventPath, aSubjectDescriptor))
+        if (MayHaveAccessibleEventPathForEndpointAndCluster(ConcreteClusterPath(aEndpoint, cluster.clusterId), aEventPath,
```

Comment:
Pending comment here: I plan to add a test for this (automated) so this does not regress and we catch devices at cert time.

Reply:
Integration test added.